fabric-rendering-v1/src/client/java/net/fabricmc/fabric/api/client/rendering/v1/HudRenderEvents.java

```diff
@@ -0,0 +1,159 @@
+	 */
+	public static final Event<Start> START = EventFactory.createArrayBacked(Start.class, listeners -> (client, context, tickCounter) -> {
+		for (Start listener : listeners) {
+			listener.onHudStart(client, context, tickCounter);
+		}
+	});
+
+	/**
+	 * Called after misc overlays (vignette, spyglass, and powder snow) have been rendered, and before the crosshair is rendered.
+	 */
+	public static final Event<AfterMiscOverlays> AFTER_MISC_OVERLAYS = EventFactory.createArrayBacked(AfterMiscOverlays.class, listeners -> (client, context, tickCounter) -> {
+		for (AfterMiscOverlays listener : listeners) {
+			listener.afterMiscOverlays(client, context, tickCounter);
+		}
+	});
+
+	/**
+	 * Called after the hotbar, status bars, experience bar, status effects overlays, and boss bar have been rendered, and before the sleep overlay is rendered.
+	 */
+	public static final Event<AfterMainHud> AFTER_MAIN_HUD = EventFactory.createArrayBacked(AfterMainHud.class, listeners -> (client, context, tickCounter) -> {
+		for (AfterMainHud listener : listeners) {
+			listener.afterMainHud(client, context, tickCounter);
+		}
+	});
+
+	/**
+	 * Called after the sleep overlay has been rendered, and before the demo timer, debug HUD, scoreboard, overlay message (action bar), and title and subtitle are rendered.
+	 */
+	public static final Event<AfterSleepOverlay> AFTER_SLEEP_OVERLAY = EventFactory.createArrayBacked(AfterSleepOverlay.class, listeners -> (client, context, tickCounter) -> {
+		for (AfterSleepOverlay listener : listeners) {
+			listener.afterSleepOverlay(client, context, tickCounter);
+		}
+	});
+
+	/**
+	 * Called after the debug HUD, scoreboard, overlay message (action bar), and title and subtitle have been rendered, and before the {@link net.minecraft.client.gui.hud.ChatHud} is rendered.
+	 */
+	public static final Event<BeforeChat> BEFORE_CHAT = EventFactory.createArrayBacked(BeforeChat.class, listeners -> (client, context, tickCounter) -> {
+		for (BeforeChat listener : listeners) {
+			listener.beforeChat(client, context, tickCounter);
+		}
+	});
+
+	/**
+	 * Called after the entire HUD is rendered.
+	 */
+	public static final Event<Last> LAST = EventFactory.createArrayBacked(Last.class, listeners -> (client, context, tickCounter) -> {
+		for (Last listener : listeners) {
+			listener.onHudLast(client, context, tickCounter);
+		}
+	});
+
+	private HudRenderEvents() { }
+
+	@FunctionalInterface
+	public interface Start {
+		/**
+		 * Called at the start of HUD rendering, right before anything is rendered.
+		 *
+		 * @param client      the {@link MinecraftClient} instance
+		 * @param context     the {@link DrawContext} instance
+		 * @param tickCounter the {@link RenderTickCounter} instance with access to tick delta
+		 */
+		void onHudStart(MinecraftClient client, DrawContext context, RenderTickCounter tickCounter);
+	}
+
+	@FunctionalInterface
+	public interface AfterMiscOverlays {
```

Comment:
Question: Are all of these interfaces the same? I do wonder if its really worth having 5 identical interfaces, this class could likely be quite a bit more consises.

Reply:
I originally had one interface but using multiple interfaces allows people to use one class to implement multiple events.